### ISSUE TITLE
segfaulting when building from_msa

### ISSUE THREAD

**Issue by iqbal-lab** [iqbal-lab-org/make_prg#57]
Here's my directory

and here's the command:

```
URI="docker://quay.io/iqballab/make_prg"

singularity exec "$URI" make_prg from_msa  -L 15 -n LASV -O pg --log outputs/L15/log.txt -f clustal clustalo-I20230528-213752-0692-38655046-p2m.clustal_num
```

output

WARNING: Cache disabled - cache location /hps/scratch/singularity is not writable.
INFO:    Downloading library image to tmp cache: /tmp/sbuild-tmp-cache-454068105
INFO:    Converting OCI blobs to SIF format
INFO:    Starting build...
2023/05/28 22:42:40  info unpack layer: sha256:a0d0a0d46f8b52473982a3c466318f479767577551a53ffc9074c9fa7035982e
2023/05/28 22:42:41  info unpack layer: sha256:70ea1bcdfee87cea0d3be42b0218fb732db46342e43b5a987332d51f67d372cc
2023/05/28 22:42:48  info unpack layer: sha256:f04e8d230c4332c19a0210f818150e1fcb2320d7abc4a20893ed0eeeb2b94734
2023/05/28 22:42:48  info unpack layer: sha256:29253eb9da6161a7e8500dc26100f98d90600c2dd1cdd6742c420ab3e364610b
2023/05/28 22:42:48  info unpack layer: sha256:a319e2b90519afce6a9b453c97eedfdfa581e8753fc9c43e38c8e5b46c249346
2023/05/28 22:42:49  info unpack layer: sha256:d28652de1e2af1c965f40a025b973bf86a480a13cf69bc3d46c740051effa13f
2023/05/28 22:42:49  info unpack layer: sha256:ad75af7e4a07bbdf41b8b262fcf2c660e585f80595ea89c3e2c0ca082bad494b
2023/05/28 22:42:49  info unpack layer: sha256:70797a5213a6145868be3b8b69eaa76d5ab8b045277d50dd423afdd381004866
INFO:    Creating SIF file...
Segmentation fault

cat outputs/L15/log.txt
INFO 28/05/2023 10:30:28 Input parameters max_nesting: 5, min_match_length: 15
INFO 28/05/2023 10:30:28 Read from MSA file clustalo-I20230528-213752-0692-38655046-p2m.clustal_num
INFO 28/05/2023 10:43:46 Input parameters max_nesting: 5, min_match_length: 15
INFO 28/05/2023 10:43:46 Read from MSA file clustalo-I20230528-213752-0692-38655046-p2m.clustal_num
INFO 28/05/2023 10:43:47 match intervals: []; non_match intervals: [[0, 3202]]

**Comment by leoisl** [comment]
For some reason I can't build containers from https://quay.io/repository/iqballab/make_prg , I am getting manifest issues. Anyway, it seems it has not created a container for the latest version, 0.4.0. I am afraid you might be using another version. Could you try using make_prg through pip or conda? You can also use the conda version through the biocontainer, running `singularity exec docker://quay.io/biocontainers/make_prg:0.4.0--pyhdfd78af_0`. I will fix this, but I think the best solution is to keep releasing pip and conda packages, and use the biocontainers. The latest container seem to work for me on the cluster:

```
$ singularity exec docker://quay.io/biocontainers/make_prg:0.4.0--pyhdfd78af_0 bash -c "cd /hps/nobackup/iqbal/leandro/temp/make_prg_test && make_prg from_msa -i clustalo-I20230528-213752-0692-38655046-p2m.clustal_num -o prg_out -O pg -f clustal -L 15"
INFO:    Using cached SIF image
2023-05-29 22:36:01.403 | INFO     | make_prg.subcommands.from_msa:run:157 - Getting input files...
2023-05-29 22:36:01.413 | INFO     | make_prg.subcommands.from_msa:run:181 - Using 1 threads to generate PRGs...
2023-05-29 22:36:01.422 | INFO     | make_prg.subcommands.from_msa:process_MSA:111 - Generating PRG for clustalo-I20230528-213752-0692-38655046-p2m.clustal_num...
```

Tell me if that worked, I will then update the containers instructions in the readme.

**Comment by mbhall88** [comment]
The build trigger was disabled because of failing builds. I had suggested waiting until you logged in and re-enabled it Leandro, but you were having login issues? Anyway, I have triggered the build for tag 0.4.0 and any new commits should build now. You should probably get your login sorted Leandro so you can also re-enable the trigger if it disables again.

There's also the biocontainer that builds for the conda recipe `docker://quay.io/biocontainers/make_prg:0.4.0--pyhdfd78af_0`

**Comment by leoisl** [comment]
Yeah, I am still getting login issues. I still get the error message: "The e-mail address is already associated with an existing Quay account. Please log in with your username and password and associate your Red Hat account to use it in the future.", even by trying to log in using my email or my username. Will try to figure this out again

**Comment by leoisl** [comment]
Something is messed up with quay.io and the red hat SSO. I think I had a quay.io account without a red hat account, and then they changed to login through red hat SSO. But when I do so, it says my email is already associated with an existing Quay account, and doesn't let me through. I made a new account, and now it works just fine. I will send you

**Comment by leoisl** [comment]
I think in the readme, at least in the first container section: https://github.com/iqbal-lab-org/make_prg#container we should point to [biocontainers](https://quay.io/repository/biocontainers/make_prg?tab=tags&tag=latest) . In another section, we can point to [our quay](https://quay.io/repository/iqballab/make_prg) . The biocontainer link is much cleaner as there we just have release containers. [our quay gets cluttered as every commit becomes a build](https://quay.io/repository/iqballab/make_prg?tab=history), and it is indeed for development, not users...

**Comment by leoisl** [comment]
@iqbal-lab have you managed to run make_prg now? Do you want me to?
